include/deal.II/numerics/vector_tools_boundary.h

```diff
@@ -95,6 +96,42 @@ namespace VectorTools
    * of these non-primitive shape functions must be @p false.
    *
    * See the general documentation of this namespace for more information.
+   *
+   * @note When solving a partial differential equation with boundary
+   *   conditions $u|_{\partial\Omega}=g$ (or on *parts* of the boundary),
+   *   then this boundary condition is in general not satisfiable exactly
+   *   using finite elements in the form $u_h|_{\partial\Omega}=g$. That is
+   *   because the function $g$ is generally not a polynomial, whereas
+   *   $u_h|_{\partial\Omega}$ *is* a polynomial on each face of the
+   *   mesh that is located at the boundary. In other words, it is in
+   *   general not possible to *impose* such boundary condition; what one
+   *   *can* do, however, is to impose
+   *     @f[ u_h|_{\partial\Omega}=I_h^{\partial\Omega} g, @f]
+   *   where $I_h^{\partial\Omega} g$ is a function that equals $g$ at each node
+   *   of the finite element space located on the boundary, and is piecewise
+   *   polynomial in between. In other words, $I_h^{\partial\Omega}$ is an
+   *   *interpolation operator* and $I_h^{\partial\Omega} g$ are the
+   *   interpolated boundary values -- thus the name. The use of
+   *   $I_h^{\partial\Omega} g$ instead of $g$ as boundary values imposes
+   *   an additional error (in the same spirit as using quadrature introduces
+   *   an additional error compared to being able to compute the integrals of
+   *   the weak form exactly). In most cases, this additional error is of the
+   *   same order as the other error terms in the finite element method,
+   *   though there are some subtle differences when measuring the error in
+   *   the $L^2$ norm. For some details, see @cite Bartels2004 .
+   *
+   * @note An alternative to using the interpolant,
+   *     @f[ u_h|_{\partial\Omega}=I_h^{\partial\Omega} g @f]
+   *   is to use the *projection* of the boundary values $g$ onto the
+   *   finite element space on the boundary:
+   *     @f[ u_h|_{\partial\Omega}=\Pi_h^{\partial\Omega} g. @f]
+   *   The projection is available using the project_boundary_values()
```

Comment:
I think it is worth pointing out that interpolation only works for nodal FEM spaces, while projection works in the more generic situation.